src/middleware.ts

```diff
@@ -1,51 +1,84 @@
-import { NextResponse } from "next/server";
-import type { NextRequest } from "next/server";
+import { match as matchLocale } from "@formatjs/intl-localematcher";
+import Negotiator from "negotiator";
+import NextAuth from "next-auth";
+import { NextRequest, NextResponse } from "next/server";
 
+import authConfig from "@/auth.config";
 import { i18n } from "@/i18n.config";
+import {
+  DEFAULT_REDIRECT_ON_LOGIN,
+  DEFAULT_REDIRECT_ON_UNAUTHORIZED,
+  apiAuthPrefix,
+  authRoutes,
+  publicRoutes,
+} from "@/routes";
 
-import { match as matchLocale } from "@formatjs/intl-localematcher";
-import Negotiator from "negotiator";
+const { auth: middleware } = NextAuth(authConfig);
 
-function getLocale(request: NextRequest): string | undefined {
-  const negotiatorHeaders: Record<string, string> = {};
-  request.headers.forEach((value, key) => (negotiatorHeaders[key] = value));
+export default middleware((request) => {
+  const { nextUrl } = request;
```

Comment:
i will check this code later, i mean looks fine but i need to better understand logic behind it.